kotest-property/src/jvmMain/kotlin/io/kotest/property/seed/seedio.kt

```diff
@@ -19,7 +19,10 @@ actual fun readSeed(path: TestPath): Long? {
    }
 }
 
-fun seedDirectory(): Path = Paths.get(System.getProperty("user.home")).resolve(".kotest").resolve("seeds")
+fun xdgCacheHomeOrNull(): String? = System.getenv("XDG_CACHE_HOME")?.takeIf { it.isNotBlank() }
+fun userHome(): String = System.getProperty("user.home")
+fun configDirectory(): String = xdgCacheHomeOrNull() ?: userHome()
+fun seedDirectory(): Path = Paths.get(configDirectory()).resolve(".kotest").resolve("seeds")
```

Comment:
```suggestion
internal fun xdgCacheHomeOrNull(): String? = System.getenv("XDG_CACHE_HOME")?.takeIf { it.isNotBlank() }
internal fun userHome(): String = System.getProperty("user.home")
internal fun configDirectory(): String = xdgCacheHomeOrNull() ?: userHome()
internal fun seedDirectory(): Path = Paths.get(configDirectory()).resolve(".kotest").resolve("seeds")
```

One thought: It might be prudent to not expose these publicly - though I haven't checked out the project locally so don't know if 'internal' is the right visibility.